cnd/src/swap_protocols/han.rs

```diff
@@ -0,0 +1,131 @@
+use crate::swap_protocols::{
+    rfc003::{
+        create_swap::{HtlcParams, SwapEvent},
+        events::{
+            Deployed, HtlcDeployed, HtlcFunded, HtlcRedeemed, HtlcRefunded, Redeemed, Refunded,
+        },
+        LedgerState,
+    },
+    state, NodeLocalSwapId, SwapId,
+};
+use chrono::NaiveDateTime;
+use futures::future::{self, Either};
+use genawaiter::{
+    sync::{Co, Gen},
+    GeneratorState,
+};
+use std::sync::Arc;
+
+/// Returns a future that tracks the swap negotiated from the given request and
+/// accept response on a ledger.
+///
+/// The current implementation is naive in the sense that it does not take into
+/// account situations where it is clear that no more events will happen even
+/// though in theory, there could. For example:
+/// - funded
+/// - refunded
+///
+/// It is highly unlikely for Bob to fund the HTLC now, yet the current
+/// implementation is still waiting for that.
+pub async fn create_watcher<C, S, L, A, H, I, T>(
+    ethereum_connector: &C,
```

Comment:
❓ Is this still restricted to Ethereum only?

Reply:
To be fixed :)